File: TrabajosDanny/Trabajos en clase/Programas_Proyecto/Menu_Cursores/Menu.cpp
#include "Menu.h"
#include <iostream>
#include <conio.h> // Para usar _getch() en la navegación de cursores.

using namespace std;

Menu::Menu() {}

void Menu::LlenarMenu() {
    int numOpciones;
    cout << "¿Cuántas opciones desea agregar al menú? ";
    cin >> numOpciones;
    cin.ignore(); // Para limpiar el buffer del `cin`.

    for (int i = 1; i <= numOpciones; ++i) {
        string opcion;
        cout << "Ingrese la opción " << i << ": ";
        getline(cin, opcion);
        menu.Insertar_cabeza(opcion); // Insertamos las opciones en la lista circular.
    }
}

void Menu::MostrarMenu() {
    cout << "Menú de opciones:\n";
    menu.Mostrar();
}

void Menu::EscogerOpcion() {
    if (menu == NULL) {
        cout << "El menú está vacío. Llene el menú antes de seleccionar una opción." << endl;
        return;
    }

    Nodo<std::string>* actual = menu.getCabeza(); // Comenzar desde la cabeza.
    char tecla;

    do {
        system("cls"); // Limpiar pantalla (en Windows).
        cout << "Use las flechas arriba/abajo para navegar y Enter para seleccionar.\n";

        // Imprimir todas las opciones con un marcador para la opción actual.
        Nodo<std::string>* aux = menu.getCabeza();
        do {
            if (aux == actual) {
                cout << "-> " << aux->getDato() << " <-\n";
            } else {
                cout << "   " << aux->getDato() << "\n";
            }
            aux = aux->getSiguiente();
        } while (aux != menu.getCabeza());

        tecla = _getch();

        if (tecla == 224) { // Detectar tecla especial.
            tecla = _getch();
            if (tecla == 72) { // Flecha arriba.
                actual = menu.getAnterior(actual);
            } else if (tecla == 80) { // Flecha abajo.
                actual = actual->getSiguiente();
            }
        }

    } while (tecla != 13); // Enter para seleccionar.

    cout << "Ha seleccionado: " << actual->getDato() << endl;
}

File: TrabajosDanny/ProgramasDePython/Listas_Python/Menu.py
import os
import msvcrt

class Menu:
    def __init__(self, titulo, opciones):
        """
        Inicializa el menú con un título y una lista de opciones.
        """
        self.titulo = titulo
        self.opciones = opciones
        self.seleccion = 0

    def mostrar(self):
        """
        Muestra el menú en pantalla, destacando la opción seleccionada.
        """
        os.system("cls" if os.name == "nt" else "clear")
        print(self.titulo)
        print("-" * len(self.titulo))
        for i, opcion in enumerate(self.opciones):
            if i == self.seleccion:
                print(f"> {opcion}")
            else:
                print(f"  {opcion}")

    def leer_tecla(self):
        """
        Lee una tecla y devuelve su código.
        """
        tecla = msvcrt.getch()
        if tecla == b'\xe0':  # Detectar teclas especiales
            tecla = msvcrt.getch()
        return tecla

    def ejecutar(self):
        """
        Ejecuta el menú y retorna la opción seleccionada.
        """
        while True:
            self.mostrar()
            tecla = self.leer_tecla()

            if tecla == b'H':  # Flecha hacia arriba
                self.seleccion = (self.seleccion - 1) % len(self.opciones)
            elif tecla == b'P':  # Flecha hacia abajo
                self.seleccion = (self.seleccion + 1) % len(self.opciones)
            elif tecla == b'\r':  # Enter
                return self.seleccion



